File: fe/src/api/auth.api.ts
import apiClient from '@/lib/api-client'
import type { ApiResponse, AuthResponse, LoginRequest, RegisterRequest } from '@/types/api'

const AUTH_BASE_URL = '/v1/api/auth'

export const authApi = {
  login: async (data: LoginRequest): Promise<AuthResponse> => {
    const response = await apiClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE_URL}/login`, data)
    return response.data.data
  },

  register: async (data: RegisterRequest): Promise<AuthResponse> => {
    const response = await apiClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE_URL}/register`, data)
    return response.data.data
  },

  refresh: async (): Promise<AuthResponse> => {
    const response = await apiClient.post<ApiResponse<AuthResponse>>(`${AUTH_BASE_URL}/refresh`)
    return response.data.data
  },

  logout: async (): Promise<void> => {
    await apiClient.post<ApiResponse<void>>(`${AUTH_BASE_URL}/logout`)
  },
}

File: be/src/main/java/com/example/spring_ecom/controller/api/auth/AuthAPI.java
package com.example.spring_ecom.controller.api.auth;

import com.example.spring_ecom.controller.api.auth.model.AuthResponse;
import com.example.spring_ecom.controller.api.auth.model.LoginRequest;
import com.example.spring_ecom.controller.api.auth.model.RegisterRequest;
import com.example.spring_ecom.core.response.ApiResponse;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("v1/api/auth")
public interface AuthAPI {

    @GetMapping("/login")
    ResponseEntity<ApiResponse<AuthResponse>> login(@Valid @RequestBody LoginRequest request);

    @GetMapping("/register")
    ResponseEntity<ApiResponse<AuthResponse>> register(@Valid @RequestBody RegisterRequest request);
}
